### PR TITLE
Updated spawning of entities to prevent stucking

### PR DESCRIPTION
**Pull request by mcNuggets1**
This pull request fixes a lot of situations, in which entities are spawned in a wall or weapons get lost because dropping in the floor.

It uses the same functions as sandbox to ensure everything spawns properly on solid surface.

If an entity has an spawn function written onto it, it will not try to unstuck it, just like in the sandbox gamemode.

### CODE REVIEW

**Comment by FPtje** [comment]
You make enough pull requests for this to start becoming annoying. Please take some time to read into branches, rebasing and other git related things. It looks like you do plenty development. Git (and other version control system) skills are a requirement if you want to do any serious work in software development.

Knowing git will help you make your personal development workflow better as well. Chunking things up into commits helps you think in terms of small and tangible milestones.

**Comment by mcNuggets1** [comment]
I'm sorry.
I'm a hobby developer, this is not yet something serious to me.

**Comment by mcNuggets1** [comment]
Well, let me try again.
I'll submit changes for the pocket weapon, so it also properly unstucks and then try to squash the two commits together.
I have downloaded Git now, so this shouldn't be a wider problem, if I don't mess everything up.

**Comment by mcNuggets1** [comment]
Did it work, @FPtje ?

**Comment by FPtje** [comment]
It says here you force pushed, so it looks like it did.

**Comment by mcNuggets1** [comment]
Well, thanks for teaching me.
Now I won't be a pain in the ass anymore, regarding pull requests.

**Comment by FPtje** [comment]
Nice work. I appreciate you taking the time for this. I'll look at the pr itself this weekend.

**Comment by mcNuggets1** [comment]
I tried staying true to the sandbox implementation.
If you however want me to use an alternate implementation, I will do so.

**Comment by mcNuggets1** [comment]
This rebasing stuff is a blessing.
Now my pull request should be complete.
I've tried making everything perfect and also found and fixed a bug, which already let to weird stucking of weapons when dropping.

**Comment by FPtje** [comment]
The functionality looks good, but I'm seeing the same code copy-paste-edited like 7 times over. That's the point where it's better to create a utility function.

**Comment by mcNuggets1** [comment]
Yeah, I tried not doing that, because I didn't know in which file I have to put it, how to name it, if I have to add it to some documentation etc.

**Comment by FPtje** [comment]
Yeah, those are things you'll need to think about. Check out `base/sv_util.lua` for examples

**Comment by mcNuggets1** [comment]
I can give it a try.

**Comment by mcNuggets1** [comment]
Don't merge yet.

**Comment by mcNuggets1** [comment]
Works now.

**Comment by mcNuggets1** [comment]
Just added hunger mod support.

**Comment by FPtje** [comment]
`DarkRP.placeEntity` is a better name, since this just spawns on top of the floor you're looking at. It doesn't do anything to stop an entity from getting stuck in a wall or platform hanging above the spawn position.

**Comment by FPtje** [comment]
Looks like github isn't seeing the merge, as I rebased.

https://github.com/FPtje/DarkRP/commit/f4a170d06d91203e254b6e490668affae3f1794c

**Comment by mcNuggets1** [comment]
Thanks for merging.
A quick question.
Can FPP be merged?
I want to update my DarkRP to the newest version today/tomorow.

**Comment by FPtje** [comment]
Done

**Comment by mcNuggets1** [comment]
Thanks!
